core/src/main/java/io/undertow/protocols/http2/HpackDecoder.java

```diff
@@ -73,8 +73,6 @@ public class HpackDecoder {
 
     private boolean first = true;
 
-    private final StringBuilder stringBuilder = new StringBuilder();
```

Comment:
There is no leakage in this class either. In this case, we are reusing the same builder across multiple decode invocations, but we are clearing the buffer right after we use it on every call. It just prevents us from having to create a new builder everytime we are invoked, which is a good thing.

Reply:
The root cause for this leak is very similar to the vulnerability CVE-2020-17527 in Tomcat. I added additional details at https://issues.redhat.com/browse/UNDERTOW-2451.